Mu2eG4/src/Mu2eG4Inputs.cc

```diff
@@ -88,6 +89,44 @@ namespace mu2e {
 
     }
       break; // SimParticles
+
+    case Mu2eG4PrimaryType::StageParticles: {
+      std::optional<art::ProductID> simid;
+
+      auto const h = evt.getValidHandle<StageParticleCollection>(primaryTag_);
+      for(const auto& s : *h) {
+        if(!simid) {
+          simid = s.parent().id();
+
+          // We have to read the pointed-to product into memory by hand
+          // for the get(ProductID, &handle) call below to work.
+          // See Kyle's e-mail on the art-users list on 2021-03-10.
+          *s.parent();
```

Comment:
It seems this line will only get called when the simid is null;  If simid has been set by an earlier loop step it will get skipped.  The comment then implies the object won't get read into memory in that case.

Reply:
There is only one collection per event, so the first pointer dereferening  will get everything into memory.